### PR TITLE
Removed non-containerized stuff from the CI pipeline.

### PR DESCRIPTION
**Pull request by shubhadapaithankar**
### Which issue this PR addresses:
Fixes : [ARO-10581](https://issues.redhat.com/browse/ARO-10581)

### What this PR does / why we need it:
This PR removes all non-containerized components from the CI pipeline as part of the effort to fully containerize and streamline the pipeline. By eliminating these non-containerized stages, the pipeline becomes more efficient, easier to maintain, and consistent with best practices for modern CI/CD workflows.

### Test plan for issue:

- Verified that all non-containerized stages are removed.
- Pipeline was tested using containerized components to ensure all stages complete successfully.

### Is there any documentation that needs to be updated for this PR?
N/A

### How do you know this will function as expected in production? 
N/A

### CODE REVIEW

File: .pipelines/ci.yml [hunk 1]
```diff
@@ -27,82 +27,6 @@ variables:
...
-              echo "##vso[task.prependpath]$(go env GOPATH)/bin"
-            displayName: Add GOBIN to path
-            target: golang
-
-          - script: |
-              set -xe
-              # Required for podman 5
-              sudo tdnf install -y gpgme-devel lvm2-devel btrfs-progs-devel golang-1.21.11-1.cm2
-              make generate
-              [[ -z "$(git status -s)" ]]
```
[the remark this answers is not in the document]
Comment (shubhadapaithankar) [reply .pipelines/ci.yml]: Kept the make generate check in the NotContainerized stage.

File: .pipelines/ci.yml [hunk 2]
```diff
@@ -27,82 +27,6 @@ variables:
...
-        pool:
-          name: 1es-aro-ci-pool
-        variables:
-          HOME: $(Agent.BuildDirectory)
-
-        steps:
-          - template: ./templates/template-checkout.yml
-          - script: |
-              set -xe
-              make test-python
```
[the remark this answers is not in the document]
Comment (shubhadapaithankar) [reply .pipelines/ci.yml:43]: yeah right. Removed the Python job from the Containerized stage and moved in the NotContainerized stage.

**Review by tsatam — APPROVED** [review]

**Review by slawande2 — APPROVED** [review]

**Pull request merged**